master/internal/api_experiment.go

```diff
@@ -448,7 +448,7 @@ func (a *apiServer) deleteExperiments(exps []*model.Experiment, userModel *model
 	taskSpec := *a.m.taskSpec
 
 	sema := make(chan struct{}, maxConcurrentDeletes)
-	wg := sync.WaitGroup{}
+	g, ctx := errgroup.WithContext(context.Background())
```

Comment:
is this ctx used anywhere? it will be canceled if there is an error, so sometimes it is clearer to just scope it to `g.Go`, hence the `errgroupx` package I added a while back.

Reply:
good catch, it is used later on in other function calls. I didn't know that it would cancel after the first time I call `g.Wait()` so that was leading to unintentional failures